app/services/katello/repository_type_manager.rb

```diff
@@ -41,29 +75,37 @@ def creatable_by_user?(repository_type)
       end
 
       def removable_content_types
-        list = @repository_types.values.map do |type|
+        list = enabled_repository_types.values.map do |type|
           type.content_types.select(&:removable)
         end
         list.flatten
       end
 
       def uploadable_content_types
-        list = @repository_types.values.map do |type|
+        list = enabled_repository_types.values.map do |type|
           type.content_types.select(&:uploadable)
         end
         list.flatten
       end
 
+      def find_defined(repository_type)
+        defined_repository_types[repository_type.to_s]
+      end
+
       def find(repository_type)
-        repository_types[repository_type.to_s]
+        found_repo_type = enabled_repository_types[repository_type.to_s]
+        if found_repo_type.blank?
+          found_repo_type = update_enabled_repository_type(repository_type.to_s)
+        end
```

Comment:
I did some performance testing and calling find() within plugin.rb was taking about ~250ms for each request.  Each call to find() was taking about ~20ms.  I think this is too much and so i dug around a bit and realized that enabled_repository_types() actually calls update_enabled_repository_type() for every single type  (even ones that are already enabled).  

So for 7 content types, and 16 calls to this method through the navidation,  this caused 7x16 = 112 calls to update_enabled_repository_type(), which i think is the expensive part.  

I modified the code to make enabled_repository_types() take an argument to turn off this 'automatic' calling of update_enabled_repository_type(), since find() is doing that already.   And make find() pass in that argument to turn it off.

This then made all the calls go down to a total of ~5 ms.  

One other thing i noticed was that enabled_repository_types() calls update_enabled_repository_type() for all repo types and not just those that are not already enabled.   I don't think there's a reason to do this, you only need to call  update_enabled_repository_type() on those that are disabled

Y